src/main/scala/org/ergoplatform/nodeView/state/UtxoStateReader.scala

```diff
@@ -84,12 +99,21 @@ trait UtxoStateReader extends ErgoStateReader with TransactionValidation[ErgoTra
     val rootHash = persistentProver.digest
     log.trace(s"Going to create proof for ${txs.length} transactions at root ${Algos.encode(rootHash)}")
     if (txs.isEmpty) {
-      Failure(new Error("Trying to generate proof for empty transaction sequence"))
+      Failure(new Exception("Trying to generate proof for empty transaction sequence"))
     } else if (!storage.version.exists(t => java.util.Arrays.equals(t, rootHash))) {
-      Failure(new Error(s"Incorrect storage: ${storage.version.map(Algos.encode)} != ${Algos.encode(rootHash)}. " +
+      Failure(new Exception(s"Incorrect storage: ${storage.version.map(Algos.encode)} != ${Algos.encode(rootHash)}. " +
         "Possible reason - state update is in process."))
     } else {
       persistentProver.avlProver.generateProofForOperations(ErgoState.stateChanges(txs).operations.map(ADProofs.changeToMod))
     }
   }
+
+  def takeSnapshot: (UtxoSnapshotManifest, Seq[UtxoSnapshotChunk]) = persistentProver.synchronized {
```

Comment:
What about performance? How much time does it take to create a snapshot for big enough state? Is it possbile to make it asynchronious?

Reply:
@catena2w, it takes about `~154ms` to create snapshot of state of size `6000000`